src/VisualStudio/Next/Remote/ServiceHubRemoteHostClient.JsonRpcSession.cs

```diff
@@ -160,6 +154,62 @@ private class SnapshotJsonRpcClient : JsonRpcClient
                     }
                 }
 
+                private async Task WriteAssetAsync(ObjectWriter writer, byte[][] checksums)
+                {
+                    // special case
+                    if (checksums.Length == 0)
+                    {
+                        await WriteNoAssetAsync(writer).ConfigureAwait(false);
+                        return;
+                    }
+
+                    if (checksums.Length == 1)
+                    {
+                        await WriteOneAssetAsync(writer, checksums[0]).ConfigureAwait(false);
+                        return;
+                    }
+
+                    await WriteMultipleAssetsAsync(writer, checksums).ConfigureAwait(false);
+                }
+
+                private Task WriteNoAssetAsync(ObjectWriter writer)
```

Comment:
yep multiple asset case can handle 0 and 1 case fine. but also, that creates bunch of temporary dictionary, set, array and etc.

special code is quite simple, so I dont think it is too much burden.